urumanifest/github.py

```diff
@@ -383,10 +386,19 @@ def _unpack_artifact(staging_path: Path, database: _WorkflowDatabase, rev: str,
         output_path.mkdir(parents=True, exist_ok=True)
 
         def iter_client_dir():
-            for i in filter(lambda x: not x.is_dir(), archive.infolist()):
+            for i in archive.infolist():
                 member_path = PurePosixPath(i.filename)
-                if len(member_path.parts) == 2 and member_path.parts[0] == "client":
-                    yield i
+                client_folder_name = "client"
+                if len(member_path.parts) >=  2 and member_path.parts[0] == client_folder_name:
+                    is_mac_app_bundle = member_path.parts[1].endswith(".app")
+                    if not i.is_dir() and len(member_path.parts) ==  2:
+                        yield ArtifactInfo(path=member_path.name, zipinfo=i, name=member_path.name)
+                    elif is_mac_app_bundle and not i.is_dir():
+                        # remove "client/"
+                        path = i.filename[len(client_folder_name)+1:]
+                        yield ArtifactInfo(path=Path(path), zipinfo=i, name=member_path.parts[1])
```

Comment:
```suggestion
                        path = member_path.relative_to(client_folder_name)
                        yield ArtifactInfo(path=path.name, zipinfo=i, name=member_path.parts[1])
```

I must admit that I'm a little confused as to why we need to remove this directory name here, though.

Reply:
The directory is multilevel so we need to recreate the directory structure and deposit everything in game_binaries. But we don't want to consider the top level client directory to be part of the directory tree being copied over.

The existing code avoided this problem because it only cared about the file name and didn't care about the path - as far as I know.

Reply:
Still looking for less manual string fiddling here.